credentialProvider/src/test/java/org/frankframework/credentialprovider/CredentialFactoryTest.java

```diff
@@ -29,4 +41,68 @@ public void testFindAliasWithPrefix() {
+		assertEquals("[alias1, account, alias2]", aliases.toString());
+	}
+
+	@Test
+	void testNoFactory() {
+		CredentialConstants.getInstance().setProperty("credentialFactory.class", "");
+		// Act
+		ICredentials c = CredentialFactory.getCredentials("account", null, null);
+
+		// Assert
+		assertNull(c.getUsername());
+		assertNull(c.getPassword());
+		assertEquals("account", c.getAlias());
+	}
+
+	@Test
+	void testMultipleFactories() {
+		// Init setting on purpose with extra whitespaces, commas etc.
+		CredentialConstants.getInstance().setProperty("credentialFactory.class", " java.util.doesNotExist , org.frankframework.credentialprovider.PropertyFileCredentialFactory,,  , org.frankframework.credentialprovider.MockCredentialFactory");
+		// Act
+		ICredentials c = CredentialFactory.getCredentials("account", null, null);
+
+		// Assert
+		assertEquals("fakeUsername", c.getUsername());
+		assertEquals("fakePassword", c.getPassword());
+	}
+
+	@Test
+	void testRightOrderMockFirst() throws Exception {
+		CredentialConstants.getInstance().setProperty("credentialFactory.class", "org.frankframework.credentialprovider.MockCredentialFactory, org.frankframework.credentialprovider.PropertyFileCredentialFactory");
+		CredentialFactory.getCredentials(null, null, null); // Make sure the factories are initialized and class loading is done
+		MockCredentialFactory.getInstance().add("account", "fakeUsername", "mockGoesFirst");
+		// Act
+		ICredentials c = CredentialFactory.getCredentials("account", null, null);
+
+		// Assert
+		assertEquals("fakeUsername", c.getUsername());
+		assertEquals("mockGoesFirst", c.getPassword());
+		assertFalse(CredentialFactory.hasCredential("notExists"));
+		assertEquals(3, CredentialFactory.getConfiguredAliases().size());
+	}
+
+	@Test
+	void testRightOrderMockLast() throws Exception {
+		CredentialConstants.getInstance().setProperty("credentialFactory.class", "org.frankframework.credentialprovider.PropertyFileCredentialFactory, org.frankframework.credentialprovider.MockCredentialFactory, ");
+		CredentialFactory.getCredentials(null, null, null); // Make sure the factories are initialized and class loading is done
+		MockCredentialFactory.getInstance().add("account", "fakeUsername", "mockGoesSecond");
+		MockCredentialFactory.getInstance().add("TheMaster", "masterUsername", "masterPassword");
+		MockCredentialFactory.getInstance().add("TheBachelor", "bachelorUsername", "bachelorPassword");
+		// Act
+		ICredentials c = CredentialFactory.getCredentials("account", null, null);
+
+		// Assert
+		assertEquals("fakeUsername", c.getUsername());
+		assertEquals("fakePassword", c.getPassword()); // comes from property file, ignore mock value
+		assertEquals("[alias1, account, alias2, TheMaster, TheBachelor]", CredentialFactory.getConfiguredAliases().toString());
+		assertTrue(CredentialFactory.hasCredential("TheMaster"));
+		assertTrue(CredentialFactory.hasCredential("fakePrefix:TheBachelor"));
+
```

Comment:
Can you add a test to see what happens when you have 2 CredentialFactories with the same alias?

Reply:
I think it was already covered by checking `account` which was in both Factories. But I've added some more accounts and data to validate it really works. Also added the check on password-only credential.